src/com/sun/jna/NativeLibrary.java

```diff
@@ -781,7 +781,8 @@ else if (Platform.isLinux() || Platform.isFreeBSD()) {
             }
         }
         else if (Platform.isAIX()) {    // can be libx.a, libx.a(shr.o), libx.so
-            if (libName.startsWith("lib")) {
+            if (isVersionedName(libName) || libName.endsWith(".so") || libName.startsWith("lib") || libName.endsWith(".a")) {
```

Comment:
In my testing I had problems loading the `jna1573645941598712075.tmp` file (which was really `libjnidispatch.a`) which I am not sure is handled here.